elytron/src/main/java/org/wildfly/extension/elytron/TlsParser.java

```diff
@@ -349,4 +355,20 @@ public void marshallSingleElement(AttributeDefinition attribute, ModelNode mappi
             .addChild(certificateAuthorityAccountParser)
             .addChild(serverSslSniContextParser)
             .build();
+
+    final PersistentResourceXMLDescription tlsParserCommunity_18_0 = decorator(TLS)
+            .addChild(decorator(KEY_STORES)
+                    .addChild(keyStoreParser)
+                    .addChild(ldapKeyStoreParser)
+                    .addChild(filteringKeyStoreParser)
+            )
+            .addChild(keyManagerParser_12_0)
+            .addChild(trustManagerParser_14_0)
+            .addChild(serverSslContextParser_9_0)
+            .addChild(clientSslContextParser_9_0)
+            .addChild(certificateAuthorityParser)
+            .addChild(certificateAuthorityAccountParser)
+            .addChild(serverSslSniContextParser)
+            .addChild(dynamicClientSslContextParser) // new
+            .build();
```

Comment:
Why not parameterize your PersistentResourceXMLDescription builder rather than duplicate this?

Reply:
@pferraro Some of these parses do not only add a new child, but also replace some child. For example look at tlsParser_14_0 which uses trustManagerParser_14_0  but the previous tlsParser_12_0 uses trustManagerParser.

Reply:
So IMO this is why we did it this way up until now. Since eg. having a previous version's builder in parameter and only adding a new child would not always work.

But I can create a subsequent jira if this should be refactored if you prefer? it does not have to be refactored in this PR

Reply:
Another one of our reasons was also to make it easier to remove older versions in the future - by duplicating we know the new version contains everything we need so we can delete the older ones with reduced risk once the time is right.

Reply:
> @pferraro Some of these parses do not only add a new child, but also replace some child. For example look at `tlsParser_14_0` which uses `trustManagerParser_14_0` but the previous `tlsParser_12_0` uses `trustManagerParser` instead.

Hmm - I'm confused.  Why would that preclude building the PersistentResourceXMLDescription dynamically based on the schema version?

Reply:
> So IMO this is why we did it this way up until now. Since eg. having a previous version's builder in parameter and only adding a new child would not always work.

Sorry if I was unclear.  The "parameter" to which I was referring is an ElytronSubsystemSchema instance, not the previous version's builder.

Reply:
Before we do anything else to refactor parsing I think we need to understand the different future updates we need to be able to support.